File: src/appVue/src/models/PaymentShare.ts
import {ShareStatus} from "@/enums/ShareStatus.ts";

export interface PaymentShare {
    user: {
        id: string;
        firstName: string;
        lastName: string;
        patronymic: string;
    };
    id: string;
    amount: number;
    share: number;
    status: ShareStatus;
}

File: CloudMining.Common/Models/Payments/Shareable/PaymentShare.cs
using CloudMining.Common.Models.Base;
using CloudMining.Common.Models.Identity;
using Modules.Payments.Domain.Enums;

namespace CloudMining.Common.Models.Payments.Shareable;

public class PaymentShare : Entity
{
	public User User { get; set; }
	public Guid UserId { get; set; }
	public Guid ShareablePaymentId { get; set; }
	public DateTime Date { get; set; }
	public decimal Amount { get; set; }
	public decimal Share { get; set; }
	public ShareStatus Status { get; set; }
}